### PR TITLE
Update and Extend Documentation

### LINKED ISSUE #11: Update and extend documentation for crate exported structures and Readme

### PR DESCRIPTION
**Pull request by markulf**
Starting with broken links. But any other kind of documentation is also welcome. 
Might by its very nature, documentation is never finished, be broken up into multiple pull requests.

### CODE REVIEW

**Comment by CPerezz** [comment]
I highly reccomend you to look at https://keepachangelog.com/en/1.0.0/ in order to add a CHANGELOG.

Anyway, since we will start on 0.1, we will just start mantaining it after that. Otherways there's A LOT of PR's to take into account.

**Comment by markulf** [comment]
> I highly reccomend you to look at https://keepachangelog.com/en/1.0.0/ in order to add a CHANGELOG.
> 
> Anyway, since we will start on 0.1, we will just start mantaining it after that. Otherways there's A LOT of PR's to take into account.

Makes sense, probably still better to have a stub there than a broken link. Added a comment that we start this process from version 0.1.

**Review by markulf — COMMENTED** [review]
The title of 3rd commit should be plonk-core/src/lib.rs

Likely these includes should just be removed and the files deleted.

File: examples/simple_circuit.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+// TODO: non-ideal hack for a first functional version.
+verify_proof::<BlsScalar, JubJubParameters, PC>(
+    &pp,
+    key,
+    &proof,
+    &public_inputs,
+    &pi_pos,
+    b"Test",
+)
+}
```
Comment (CPerezz) [inline examples/simple_circuit.rs]: Add new line at the end of the file

File: examples/simple_circuit.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+//
+// Copyright (c) ZK-GARAGE. All rights reserved.
+
+//! PLONK Example
+
+use ark_bls12_381::{Bls12_381, Fr as BlsScalar};
+//use ark_ec::{PairingEngine, TEModelParameters};
```
Comment (CPerezz) [inline examples/simple_circuit.rs]: Remove the commented stuff

**Review by CPerezz — COMMENTED** [review]
Looks good. 

We might add a bit simpler examples IMO. Or group them by:
- Arith
- Bool
- ECC
- Range
- Misc

**Comment by markulf** [comment]
Closing this. Too many small building sites for one pull request.

**Review by LukePearson1 — APPROVED** [review]
LGTM! 

Had to so some merging to get up to date with the latest branch.

File: examples/simple_circuit.rs [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+use ark_ec::models::twisted_edwards_extended::GroupAffine;
+use ark_ec::{AffineCurve, ProjectiveCurve};
+use ark_ed_on_bls12_381::{
+    EdwardsParameters as JubJubParameters, Fr as JubJubScalar,
+};
+use plonk_core::circuit::{verify_proof, Circuit, PublicInputBuilder};
+use plonk_core::constraint_system::StandardComposer;
+use plonk_core::error::Error;
+
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
```
Comment (bhgomes) [inline examples/simple_circuit.rs]: @LukePearson1 still need to remove this line

**Review by bhgomes — CHANGES_REQUESTED** [review]

**Review by bhgomes — APPROVED** [review]

**Comment by CPerezz** [comment]
@bhgomes @LukePearson1 This PR started to make the `intra-doc links` fail. 

Why was it merged?

**Comment by bhgomes** [comment]
> @bhgomes @LukePearson1 This PR started to make the `intra-doc links` fail. 
> 
> 
> 
> Why was it merged?

We don't know how the intra-doc links are broken since this PR doesn't seem to change anything related to that.

**Pull request merged**